src/sentry/event_manager.py

```diff
@@ -756,31 +757,38 @@ def _get_or_create_release_many(jobs, projects):
             release_date_added[release_key] = new_datetime
 
     for (project_id, version), jobs_to_update in jobs_with_releases.items():
-        release = Release.get_or_create(
-            project=projects[project_id],
-            version=version,
-            date_added=release_date_added[(project_id, version)],
-        )
+        try:
+            release = Release.get_or_create(
+                project=projects[project_id],
+                version=version,
+                date_added=release_date_added[(project_id, version)],
+            )
+        except ValidationError:
+            release = None
+            logger.exception(
+                "Failed creating Release(project=%s, version=%s)", projects[project_id], version
+            )
```

Comment:
I don't know if it's still the case, but parameterising messages can sometimes cause events not to group correctly.

You're better off passing these as an `extra` dict